internal/pkg/deploy/cloudformation/app.go

```diff
@@ -129,6 +129,66 @@ func (cf CloudFormation) upgradeAppStack(conf *stack.AppStackConfig) error {
 	return cf.executeAndRenderChangeSet(in)
 }
 
+// removeDNSDelegationAndCrossAccountAccess removes the provided account ID from the list of accounts that can write to the
+// application's DNS HostedZone. It does this by creating the new list of DNS delegated accounts, updating the app
+// infrastructure roles stack, then redeploying all the stackset instances with the new list of accounts.
+// If the list of accounts already excludes the account to remove, we return early for idempotency.
+func (cf CloudFormation) removeDNSDelegationAndCrossAccountAccess(appStack *stack.AppStackConfig, accountID string) error {
+	// Get the most recently deployed list of delegated accounts.
+	appStackDesc, err := cf.cfnClient.Describe(appStack.StackName())
+	if err != nil {
+		return fmt.Errorf("get existing application infrastructure stack: %w", err)
+	}
+	dnsDelegatedAccounts := cf.dnsDelegatedAccountsForStack(appStackDesc.SDK())
+
+	// Remove the desired account from this list.
+	var newAccountList []string
+	for _, account := range dnsDelegatedAccounts {
+		if account == accountID {
+			continue
+		}
+		newAccountList = append(newAccountList, account)
+	}
+	// If the lists are equal length, the account has already been removed and we don't have to redeploy the stackset instances.
+	if len(newAccountList) == len(dnsDelegatedAccounts) {
+		return nil
+	}
+	// Create a new AppStackConfig using the new account list.
+	newCfg := stack.NewAppStackConfig(&deploy.CreateAppInput{
+		Name:                  appStack.Name,
+		AccountID:             appStack.AccountID,
+		DNSDelegationAccounts: newAccountList,
+		DomainName:            appStack.DomainName,
+		DomainHostedZoneID:    appStack.DomainHostedZoneID,
+		PermissionsBoundary:   appStack.PermissionsBoundary,
+		AdditionalTags:        appStack.AdditionalTags,
+		Version:               appStack.Version,
+	})
+	// Redeploy the infrastructure roles stack.
+	s, err := toStack(newCfg)
+	if err != nil {
+		return err
+	}
+	// Update app stack.
+	if err := cf.cfnClient.UpdateAndWait(s); err != nil {
+		var errNoUpdates *cloudformation.ErrChangeSetEmpty
+		if errors.As(err, &errNoUpdates) {
+			return nil
+		}
+		return fmt.Errorf("update application to remove DNS delegation from account %s: %w", accountID, err)
+	}
+	// Update stackset instances to remove account.
+	appResourcesConfig, err := cf.getLastDeployedAppConfig(appStack)
+	if err != nil {
+		return err
+	}
+	appResourcesConfig.Accounts = newAccountList
+	if err := cf.deployAppConfig(newCfg, appResourcesConfig, true); err != nil {
```

Comment:
Callout not for this PR but we probably need to refactor `deployAppConfig` to avoid using any boolean value (maybe split into two)

Reply:
Quick question! Why do we need to call `deployAppConfig` which looks like is updating the stackset? 

My understanding so far is that stackset is only touched when we add/delete new regions & add/delete new workloads. Why do we need to update stackset when we remove a account?

Reply:
oh I think I see now :homer-disappearing

We need to update the app stack so that we remove the account from delegation role; we need to update the stackset so that we remove the account from the metadata.